pkg/operator/ceph/pool/controller.go

```diff
@@ -295,6 +295,7 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 				logger.Debug("external rgw endpoint monitoring go routine already running!")
 			} else {
 				checker := newMirrorChecker(r.context, r.client, r.clusterInfo, request.NamespacedName, &cephBlockPool.Spec, cephBlockPool.Name)
+				r.blockPoolChannels[cephBlockPool.Name].monitoringRunning = true
```

Comment:
So this is breaking the CI since we have 2 clusters in different namespaces with the same name. It's not super realistic though. I don't think we need to change the name of the `blockPoolChannels` key. Could you change the name of the second CephBlockPool CRD in the rook-ceph-secondary namespace here: https://github.com/rook/rook/blob/master/.github/workflows/canary-integration-test.yml#L742 and call it replicapool2 or something?

Reply:
If someone has two ceph clusters, it actually seems likely that they would have pools with the same name in both of them. For example, `replicapool` is used in all the examples and it is rarely changed as the default name. It seems like we need a follow up PR for this.

Reply:
can this be `clusterNamespace+poolName` or `poolNamespace+poolName` which makes it unique?

Reply:
I'm still not certain this is a common scenario, but let's play it safe and incorporate the namespace in the `blockPoolChannels` key as well.

Reply:
CI looks good now.